modules/pool/lib/apr-data-sources/yb-tokens-apr.service.ts

```diff
@@ -72,10 +82,21 @@ export class YbTokensAprService implements PoolAprService {
 
                 let userApr = tokenApr.apr * tokenPercentageInPool;
 
-                if (collectsYieldFee(pool) && tokenCollectsYieldFee(token) && token.dynamicData) {
+                // AAVE + LST case, we need to apply the underlying token APR on top of the AAVE market APR
+                const aaveUnderlying = this.aaveUnderlyingMap[token.address];
+                if (aaveUnderlying) {
+                    const underlyingTokenApr = aprs.get(aaveUnderlying);
+                    if (underlyingTokenApr) {
+                        userApr = ((1 + tokenApr.apr) * (1 + underlyingTokenApr.apr) - 1) * tokenPercentageInPool;
+                    }
+                }
+
+                if (collectsYieldFee(pool) && tokenCollectsYieldFee(token) && pool.dynamicData) {
                     const fee =
                         pool.type === 'META_STABLE'
                             ? parseFloat(pool.dynamicData.protocolSwapFee || '0')
+                            : pool.protocolVersion === 3
+                            ? parseFloat(pool.dynamicData.aggregateYieldFee || '0.1')
```

Comment:
This also needs to changed in the networkconfig from 0.5 to 0.1 for v3. Also, why is it called aggregateYieldFee and not protocolYieldFee for v3 pools?

Reply:
AggregateYieldFee includes potential hook fees. We might not need a default since we always query for it on-chain. Do you agree?

Reply:
I’m not sure about aggregateYieldFee vs. protocolYieldFee. I assume protocolYieldFee can be added via the protocolFeeController, but most usage will likely remain with aggregateYieldFee.

Reply:
added: https://github.com/balancer/backend/issues/1225